src/app/space/forge-wizard/abstract-wizard.component.ts

```diff
@@ -0,0 +1,301 @@
+    this.context.current.subscribe((ctx: Context) => {
+      if (ctx.space) {
+        this.currentSpace = ctx.space;
+        console.log(`ForgeWizardComponent::The current space has been updated to ${this.currentSpace.attributes.name}`);
+      }
+    });
+  }
+
+  ngOnInit(): void {
+  }
+
+  get currentGui(): Gui {
+    return this.history.currentGui;
+  }
+
+  cancel($event) {
+    this.onCancel.emit($event);
+  }
+
+  abstract executeStep(steps: WizardStep[]);
+
+  reviewStep(): void {
+    this.isLoading = true;
+    let space = this.currentSpace;
+    let codebases: Codebase[] = this.convertResultToCodebases(this.result);
+    let obs: Observable<Codebase>;
+    codebases.forEach(code => {
+      if (!obs) {
+        obs = this.addCodebaseDelegate(space.id, code);
+      } else {
+        obs = obs.concat(this.addCodebaseDelegate(space.id, code));
+      }
+    });
+    obs.subscribe(
+      codebase => {
+        console.log(`Successfully added codebase ${codebase.attributes.url}`);
+        // todo broadcast
+        // this._broadcaster.broadcast('codebaseAdded', codebase);
+        this.notifications.message(<Notification>{
+          message: `Your ${codebase.attributes.url} repository has been `
+          + `added to the ${this.currentSpace.attributes.name} space`,
+          type: NotificationType.SUCCESS
+        });
+      },
+      err => console.log(`Error adding codebase ${err}`),
+      () => {
+        console.log(`completed`);
+        this.isLoading = false;
+        // TODO Display error
+        this.onCancel.emit({});
+      });
+  }
+
+  nextClicked($event: WizardEvent): void {
+    if (this.history.stepIndex === this.EXECUTE_STEP_INDEX + 1) { // execute
+      this.executeStep(flattenWizardSteps(this.wizard));
+    } else if (this.history.stepIndex === this.LAST_STEP + 1) { // addcodebaseStep
+      this.reviewStep();
+    } else { // init or next
+      this.loadUi().catch(error => {
+        this.isLoading = false;
+        this.error = error;
+      });
+    }
+  }
+
+  previousClicked($event: WizardEvent): void {
+    // history step index starts at index 1
+    this.move(this.history.stepIndex - 1, this.history.stepIndex - 2, flattenWizardSteps(this.wizard));
+  }
+
+  stepChanged($event: WizardEvent) {
+    const currentStep = this.history.stepIndex;
+    const gotoStep = $event.step.config.priority;
+    if (currentStep !== this.LAST_STEP + 1 && currentStep > gotoStep) {
+      this.move(currentStep - 1, gotoStep - 1, flattenWizardSteps(this.wizard));
+    }
+  }
+
+  // to, from are zero-based index
+  move(from: number, to: number, wizardSteps = this.wizard.steps) {
+    let serverSideErrors = null;
+    if (to === this.EXECUTE_STEP_INDEX) { // last forge step, change next to finish
+      this.wizard.config.nextTitle = 'Finish';
+    }
+    if (from === this.EXECUTE_STEP_INDEX && from > to) { // moving from finish step to previous, set back next
+      this.wizard.config.nextTitle = '> Next';
```

Comment:
Wherever the back/next buttons are used, it should use `<span class="i fa fa-angle-right"></span>` or `<span class="i fa fa-angle-left"></span>` and the format for next should be "Next >" with the caret following the word.

Reply:
@srambach I've changed the name of the button but for the css I think this is a PR for https://github.com/patternfly/patternfly-ng/blob/master/src/app/wizard/wizard.component.html#L51